pkg/storage/ingest/pusher.go

```diff
@@ -24,21 +24,28 @@ import (
 	"github.com/grafana/mimir/pkg/util/spanlogger"
 )
 
+const shardForSeriesBuffer = 2000 // TODO dimitarvdimitrov 2000 is arbitrary; the idea is that we don't block the goroutine calling PushToStorage while we're flushing. A linked list with a sync.Cond or something different would also work
```

Comment:
I thought a bit about this - i think it shouldn't be larger than 1 or 2. 

We want to batch and flush in parallel. There's no point in batching faster than we can flush, right? If we flush 1 batch/sec, then batching 2 batches/sec doesn't make us faster. So if we're still flushing batch `N`, batch `N+1` is enqueued for flushing and we've just finished buffering batch `N+2` we're going too fast - in the time to flush `N` we've created `N+2` and `N+1`.

So I think a buffer of 1 or 2 is enough. What do you think?

Reply:
forgot to mention that we don't have to finish this discussion in this PR. Just saw the line changed and decided to bring it up

Reply:
I haven't thought too much about this yet; I'll put this discussion for the next PR if you don't mind.